### PR TITLE
Add Marten web framework

### PR DESCRIPTION
**Pull request by ellmetha**
# Description

This pull request adds [Marten](https://martenframework.com) to the benchmarked Crystal web frameworks. Please let me know if I am missing any configuration files! 🙏

### CODE REVIEW

File: crystal/marten/config.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+framework:
+  website: martenframework.com
+  github: martenframework/marten
+  version: 0.4.0
```
Comment (waghanza) [inline crystal/marten/config.yml:4]:
```suggestion
  version: 0.4
```

**Review by waghanza — COMMENTED** [review]
Thanks @ellmetha for your contribution, a small feedback tho

File: crystal/marten/config/routes.cr [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Marten.routes.draw do
+  path "/", RootHandler, name: "root"
```
[the remark this answers is not in the document]
Comment (ellmetha) [reply crystal/marten/config/routes.cr:2]: Probably not! In Marten, there is a strong assumption that the routing mechanism should only care about request paths. 🙂 

The decision to respond to certain request methods or not is left to the actual [handler](https://martenframework.com/docs/handlers-and-http/introduction#writing-handlers) who can decide to implement corresponding methods (such as `#get`, `#post`, etc). If a handler does not implement a dispatching logic for a given HTTP method, then a "Not allowed" response (405) response is returned automatically by the handler.

[the remark this answers is not in the document]
Comment (waghanza) [reply crystal/marten/config/routes.cr:2]: is it not a security issue @ellmetha ? I mean responding a 405 could be a part of API discovery for an attacker, I would suggest a 404

File: crystal/marten/config.yml
Comment (waghanza) [inline crystal/marten/config.yml:1]: sorry, should be `crystal/marten/config.yaml` instead ;-)


CI will trigger if so

**Review by waghanza — CHANGES_REQUESTED** [review]

File: crystal/marten/config.yml
Comment (ellmetha) [reply crystal/marten/config.yml:1]: Ah gotcha!

File: crystal/marten/config/routes.cr (hunk 2 above)
[the remark this answers is not in the document]
Comment (ellmetha) [reply crystal/marten/config/routes.cr:2]: That's an interesting point. I know that some frameworks have the same mechanism as Marten regarding this aspect (this is the case of Django for example). But making it possible to configure this is something that could be beneficial I think.

[the remark this answers is not in the document]
Comment (waghanza) [reply crystal/marten/config/routes.cr:2]: Also having this into the routing part allows to have a more descriptive file, having at a blink of an eye the existing routes (I mean verbs and URLs)

**Review by waghanza — APPROVED** [review]

**Comment by waghanza** [comment]
@ellmetha results are updated https://web-frameworks-benchmark.netlify.app/result?l=crystal, but `marten` seems very low

**Pull request merged**